frictionless/fields/year.py

```diff
@@ -25,6 +26,10 @@ def value_reader(cell):
                     return None
                 if len(cell) != 4:
                     return None
+                # for numeric string starting with zero for example, 001, 00, 01
+                has_leading_zero = re.match(r"0\d+", cell.strip())
```

Comment:
I think we should allow leading zeroes for `year` as the standards say that it should be in `YYYY` format - https://specs.frictionlessdata.io/table-schema/#year

So I think we just need to apply it for integer/number

Reply:
But for the example shared in the issue has a field "codice_comune" which I think is also a code but has the pattern of YYYY. The field gets inferred as year. But when I extract data, the zeros gets stripped:

- name: codice_comune
      type: year

`frictionless extract "https://gist.githubusercontent.com/aborruso/076e5fad847b658a535b16cbcf3abdfd/raw/887edc88ab75b01d15ea1ac4bb052ffaf8d5ef9c/tmp.csv"`

`codice_comune 
1360   
530
580
870
90

What do we do for such kind of field?